### PR TITLE
Object Model API

### PR DESCRIPTION
**Pull request by singhsarab**
Changes based on API scan.

### CODE REVIEW

File: src/Microsoft.TestPlatform.ObjectModel/Constants.cs [hunk 1]
```diff
@@ -68,6 +68,11 @@ public static class Constants
         public static readonly Architecture DefaultPlatform = XmlRunSettingsUtilities.OSArchitecture == Architecture.ARM ? Architecture.ARM : Architecture.X86;
 
         /// <summary>
+        /// Adding this for compatibility
+        /// </summary>
+        public const FrameworkVersion DefaultFramework = FrameworkVersion.Framework40;
```
Comment (codito) [inline src/Microsoft.TestPlatform.ObjectModel/Constants.cs]: `[Deprecated]`

File: test/Microsoft.TestPlatform.Common.UnitTests/Microsoft.TestPlatform.Common.UnitTests.csproj [hunk 2]
```diff
@@ -7,7 +7,7 @@
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
     <OutputType Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">Exe</OutputType>
-    <TargetFrameworks>netcoreapp1.0;net451</TargetFrameworks>
+    <TargetFrameworks>net451;netcoreapp1.0</TargetFrameworks>
```
Comment (codito) [inline test/Microsoft.TestPlatform.Common.UnitTests/Microsoft.TestPlatform.Common.UnitTests.csproj]: Why this change?

File: src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs [hunk 3]
```diff
@@ -309,6 +314,25 @@ public Framework TargetFrameworkVersion
...
+        /// </summary>
+        [Obsolete("Use TargetFramework instead")]
+        public FrameworkVersion TargetFrameworkVersion
+        {
+            get
+            {
+                return this.frameworkVersion;
+            }
+
+            set
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs]: `set_TargetFramework` should set `frameworkVersion`.
